src/link/Elf.zig

```diff
@@ -1650,9 +1650,15 @@ fn linkWithLLD(self: *Elf, comp: *Compilation) !void {
                 if (self.base.options.libc_installation != null) {
                     const needs_grouping = self.base.options.link_mode == .Static;
                     if (needs_grouping) try argv.append("--start-group");
-                    try argv.append("-lm");
-                    try argv.append("-lpthread");
-                    try argv.append("-lc");
+                    // This matches the order of glibc.libs
+                    try argv.appendSlice(&[_][]const u8{
+                        "-lm",
+                        "-lpthread",
+                        "-lc",
+                        "-ldl",
+                        "-lrt",
+                        "-lutil",
```

Comment:
it is breaking openbsd stage2 build : several libraries (`dl` and `rt`) doesn't exist on OpenBSD, and `util` isn't need

Reply:
Hrm, we could really use OpenBSD coverage in the CI. I guess we need to add information about what libc components each target supports to `std.Target`.

Reply:
@semarie I've opened #8754 to fix the build for now.

I'm also considering ways we can improve on this situation and not link libc components that aren't required .